pkg/apis/garden/validation/validation.go

```diff
@@ -182,17 +185,36 @@ func ValidateCloudProfileSpec(spec *garden.CloudProfileSpec, fldPath *field.Path
 		allErrs = append(allErrs, validateZones(spec.OpenStack.Constraints.Zones, fldPath.Child("openstack", "constraints", "zones"))...)
 
 		floatingPoolPath := fldPath.Child("openstack", "constraints", "floatingPools")
-		if len(spec.OpenStack.Constraints.FloatingPools) == 0 {
```

Comment:
Can you add tests for these cases in `pkg/garden/validation/validation_test.go`?